streaming/src/main/scala/org/apache/spark/streaming/dstream/PairDStreamFunctions.scala

```diff
@@ -350,6 +349,18 @@ class PairDStreamFunctions[K, V](self: DStream[(K, V)])
     )
   }
 
+  /** TODO: Add scala docs */
+  def trackStateByKey[S: ClassTag, T: ClassTag](
```

Comment:
For the method signature, since `spec` contains `ClassTag`s for `S` and `T`, we can get it from `spec` and don't need to add `ClassTag` in the method signature.